### DISCUSSION TITLE
SetSubVector

### DISCUSSION THREAD

**Discussion by 0Ger** [mathnet/mathnet-numerics#824] [Q&A] [answered]
Hello,
I found that he SetSubVector(int start, int length, Vector values) is not working properly. 
The code was somethin like

Matrix<double> A .... init A(m,n)
Vector<double>b ... init b(m-1)
A.Column(j).SetSubVector(1,m-1, b);

The Subvector of A did not change. Only a workaround using for loop had the correct result
                    for(int h = 1; h<m;h++)
                    {
                        A[h, j] = hlp[h - 1];
                    }

**Comment by cdrnet** [comment, marked as answer]
Note that the column vector returned by `.Column(j)` is independent from A. So `SetSubVector` operates on that column vector instead of A. It seems that you could use `SetColumn` instead:

    A.SetColumn(j,1,m-1,b);

**Comment by 0Ger** [comment]
Thank you,
I did not realize that  A.Column(j) is a copy.
